fdbrpc/include/fdbrpc/FileTransfer.h

```diff
@@ -0,0 +1,259 @@
+/**
+ * FileTransfer.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifdef FLOW_GRPC_ENABLED
+#ifndef FDBRPC_FILE_TRANSFER_H
+#define FDBRPC_FILE_TRANSFER_H
+
+#include <cstdio>
+#include <optional>
+#include <fstream>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/server_context.h>
+#include <grpcpp/support/status.h>
+
+#include <boost/asio.hpp>
+
+#include "flow/IRandom.h"
+#include "crc32/crc32c.h"
+#include "fdbrpc/file_transfer/file_transfer.pb.h"
+#include "fdbrpc/file_transfer/file_transfer.grpc.pb.h"
+
+using fdbrpc::DownloadChunk;
+using fdbrpc::DownloadRequest;
+using fdbrpc::FileTransferService;
+using fdbrpc::GetFileInfoReply;
+using fdbrpc::GetFileInfoRequest;
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::Status;
+
+static uint32_t crc32_checksum_ifstream(std::ifstream* input_file);
+
+class FileTransferServiceImpl final : public FileTransferService::Service {
```

Comment:
TODO:

- gRPC should respect FDB TLS settings
- Security considerations. Limit what files/folders can be downloaded when instantiating this.